pkg/querier/queryrange/index_stats_cache.go

```diff
@@ -0,0 +1,79 @@
+package queryrange
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/log"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
+	"github.com/grafana/loki/pkg/storage/chunk/cache"
+	"github.com/grafana/loki/pkg/util"
+)
+
+type IndexStatsSplitter struct {
+	Limits
+}
+
+// GenerateCacheKey generates a cache key based on the userID, Request and interval.
+func (i IndexStatsSplitter) GenerateCacheKey(_ context.Context, userID string, r queryrangebase.Request) string {
```

Comment:
Can we reuse the following, which has the same implementation?
```go
type cacheKeyLimits struct {
	Limits
	transformer UserIDTransformer
}
```

Reply:
Done. Similarly to [what the log results cache does (uses `log` prefix)][1], the idea was to add the `indexStats` prefix to the key to distinguish them from other queries (e.g. series queries, or log queries (w/o filter)),. Looks like the `NewSeriesTripperware` nor the `NewLimitedTripperware` configure any cache, so we should be safe there. Still, I think it would be a good idea to add the `indexStats` prefix just in case we decide to implement these caches in the future.

So instead of using the `cacheKeyLimits` directly, I'm  doing the following:
```go
type IndexStatsSplitter struct {
	cacheKeyLimits
}

// GenerateCacheKey generates a cache key based on the userID, Request and interval.
func (i IndexStatsSplitter) GenerateCacheKey(ctx context.Context, userID string, r queryrangebase.Request) string {
	cacheKey := i.cacheKeyLimits.GenerateCacheKey(ctx, userID, r)
	return fmt.Sprintf("indexStats:%s", cacheKey)
}
```

[1]: https://github.com/grafana/loki/blob/9159c1dac3ad1e80b9ebf0fe256dbe6dc0b8e903/pkg/querier/queryrange/log_result_cache.go#L125